ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-form.collection.js

```diff
@@ -39,16 +40,23 @@ module.exports = Backbone.AssociatedModel.extend({
   defaults: {
     doneLoading: false,
     added: false,
-    resultForms: [],
-  },
-  initialize: function() {
-    this.addResultForm(
+    resultForms: [
       new ResultForm({
         title: 'Create New Data View',
         type: 'new-result',
-      })
-    )
-    this.addResultForms()
+      }),
+    ],
+  },
+  initialize: function() {
+    if (promiseIsResolved === true) {
+      this.addResultForms()
+      promiseIsResolved = false
+      bootstrapPromise = new resultTemplatePromise()
+    }
+    bootstrapPromise.then(() => {
+      this.addResultForms()
```

Comment:
Why are we executing `this.addResultForms()` here and on line 52?

Reply:
I was refactoring this file to more closely follow `search-form.collection.js`